rust/relay/ebpf-turn-router/src/main.rs

```diff
@@ -451,6 +448,11 @@ fn handle_ipv4_udp_to_ipv4_channel(
     let new_ipv4_dst = client_and_channel.client_ip();
     let new_ipv4_len = old_ipv4_len + CdHdr::LEN as u16;
 
+    // Check for packet loop - would we be sending to ourselves?
+    if new_ipv4_src == new_ipv4_dst {
+        return Err(Error::PacketLoop);
+    }
+
```

Comment:
I'm not sure if this is intended behavior, but I noticed that we generate relay-relay candidates for a relay to talk to itself. That can cause weird packet loops and is generally not needed, at least in the eBPF code path, so I implemented a simple loop detection in the eBPF module to prevent these things.